torch/fx/experimental/fuser.py

```diff
@@ -62,3 +70,279 @@ def fuse(model: torch.nn.Module, inplace=False) -> torch.nn.Module:
+            if type(cur_module) == nn.Dropout:
+                prv_node = node.args[0]
+                node.replace_all_uses_with(prv_node)
+                new_graph.erase_node(node)
+    new_graph.lint()
+    result = fx.GraphModule(fx_model, new_graph)
+    return result
+
+
+
+def extract_subgraph(orig_module: nn.Module, nodes: List[fx.Node], inputs: List[fx.Node], outputs: List[str]):
+    """
+    Given lists of nodes from an existing graph that represent a subgraph, returns a submodule that executes that subgraph.
+    """
+    new_graph = fx.Graph()
+    env = {}
+    for input in inputs:
+        new_node = new_graph.placeholder(input.name)
+        env[input.name] = new_node
+    for node in nodes:
+        new_node = new_graph.node_copy(node, lambda x: env[x.name])
+        env[node.name] = new_node
+    new_graph.output([env[output] for output in outputs])
+    new_graph.lint()
+    return fx.GraphModule(orig_module, new_graph)
+
+mkldnn_supported = [
+    nn.Conv2d, nn.Linear, nn.BatchNorm2d, nn.ReLU, nn.MaxPool2d, nn.AvgPool2d, nn.AdaptiveAvgPool2d,
+    torch.relu, torch.transpose,
+    F.relu, F.avg_pool2d, F.adaptive_avg_pool2d
+]
+mkldnn_supported_unknown = [operator.add, operator.mul]
+mkldnn_map = {
+    nn.Conv2d: th_mkldnn.MkldnnConv2d,
+    nn.Linear: th_mkldnn.MkldnnLinear,
+    nn.BatchNorm2d: lambda a, b: th_mkldnn.MkldnnBatchNorm(a)
+}
+
+old_modules: Dict[nn.Module, nn.Module] = {}
+def modules_to_mkldnn(nodes: List[fx.Node], modules: Dict[str, nn.Module]):
+    """
+    For each node, if it's a module that can be preconverted into MKLDNN,
+    then we do so and create a mapping to allow us to convert from the MKLDNN
+    version of the module to the original.
+    """
+    for node in nodes:
+        if node.op == 'call_module':
+            assert(isinstance(node.target, str))
+            cur_module = modules[node.target]
+            if type(cur_module) in mkldnn_map:
+                new_module = mkldnn_map[type(cur_module)](cur_module, torch.float)
+                assert(isinstance(new_module, nn.Module))
+                old_modules[new_module] = copy.deepcopy(cur_module)
+                replace_node_module(node, modules, new_module)
+
+def reset_modules(nodes: List[fx.Node], modules: Dict[str, nn.Module]):
+    """
+    Maps each module that's been changed with `modules_to_mkldnn` back to its
+    original.
+    """
+    for node in nodes:
+        if node.op == 'call_module':
+            assert(isinstance(node.target, str))
+            cur_module = modules[node.target]
+            if cur_module in old_modules:
+                replace_node_module(node, modules, old_modules[cur_module])
+
+class MklSubgraph:
+    def __init__(self, fx_graph: fx.Graph):
+        self.fx_graph = fx_graph
+        self.nodes: List[fx.Node] = []
+        self.start_nodes: List[fx.Node] = []
+        self.end_nodes: List[fx.Node] = []
+
+def gen_mkl_autotuner(example_inputs, iters=10, warmup=1):
```

Comment:
This should have a docstring explaining that it's a heuristic you can pass in to `prepare_for_inference`, and it should explain what this heuristic does